exchanges/oex/oex_wrapper.go

```diff
@@ -0,0 +1,390 @@
+package oex
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/websocket/wshandler"
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+// Start starts the Oex go routine
+func (o *Oex) Start(wg *sync.WaitGroup) {
+	wg.Add(1)
+	go func() {
+		o.Run()
+		wg.Done()
+	}()
+}
+
+// Run implements the Oex wrapper
+func (o *Oex) Run() {
+	if o.Verbose {
+		log.Debugf("%s polling delay: %ds.\n", o.GetName(), o.RESTPollingDelay)
+		log.Debugf("%s %d currencies enabled: %s.\n", o.GetName(), len(o.EnabledPairs), o.EnabledPairs)
+	}
+	exchangeCurrencies, err := o.GetAllPairs()
+	if err != nil {
+		log.Errorf("%s Failed to get available symbols.\n", o.GetName())
+	}
+	var newExchangeCurrencies currency.Pairs
+	var tempPair currency.Pair
+	for x := 0; x < len(exchangeCurrencies.Data); x++ {
+		tempPair = currency.NewPairFromString(exchangeCurrencies.Data[x].Symbol)
+		newExchangeCurrencies = append(newExchangeCurrencies, exchange.FormatCurrency(tempPair))
+	}
+	err = o.UpdateCurrencies(newExchangeCurrencies, false, true)
+	if err != nil {
+		log.Errorf("%s Failed to update available currencies %s.\n", o.GetName(), err)
+	}
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (o *Oex) UpdateTicker(p currency.Pair, assetType string) (ticker.Price, error) {
+	var resp ticker.Price
+
+	strPair := exchange.FormatExchangeCurrency(o.Name, p).String()
+	tempResp, err := o.GetAllTickers()
+	if err != nil {
+		return resp, err
+	}
+	for a := range tempResp.Data.Ticker {
+		if tempResp.Data.Ticker[a].Symbol != strPair {
```

Comment:
Since this exchange supports fetching all tickers, it's best to call o.GetEnabledCurrencies() and match them up to the ticker response, then finally process all tickers using ticker.ProcessTicker(...). You can then return the result of ticker.GetTicker for the supplied pair. This allows for 1 call of UpdateTicker to process all enabled tickers in one request.